pkg/wait/conditions.go

```diff
@@ -31,11 +31,11 @@ type Range struct {
 }
 
 // WaitCondition is a function performing a condition check for if we need to keep waiting
-type WaitCondition func() (done bool, err error)
+type WaitCondition func() (done bool, err error) // nolint
 
-// clusterApiObject is an interface that combines controller-runtime Object and Cluster API object with conditions.
+// clusterAPIObject is an interface that combines controller-runtime Object and Cluster API object with conditions.
 // We use this in functions where Kubernetes client fetches Cluster API objects and checks their Status.Conditions.
-type clusterApiObject interface {
+type clusterAPIObject interface {
```

Comment:
There was some requirement renaming the interface from golangci-lint, those might have an impact. I can also ignore it if needed